woocommerce/woocommerce-ios
# HACK week: Replace WPKit Announcement with another model in the app to remove the dependency

Part 2/3 of the HACK week project to remove WPKit dependency from the app.


## Description

This PR adds models, mapper and remote to handle feature announcements within the app. This allows us to update `AnnouncementStore` to use the in-house implementation instead of importing `WordPressKit`.

With this change, we can remove `WordPressKit` from the Yosemite layer 🎉 

## Testing instructions

There are no announcements for the current release, so you'll need to update `AnnouncementStore` to set `appVersion` to 10.7:

<img width="934" src="https://user-images.githubusercontent.com/5533851/223416673-55e55a46-4bbf-403e-b037-63fca03a8a68.png">

- Build and run the app.
- Log in if needed.
- Confirm the "What's New" modal appears when the app launches.

## Screenshots

<img src="https://user-images.githubusercontent.com/5533851/223418578-69553808-bcbc-439c-8f02-db575b880270.png" width=320 />


---
